torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -712,7 +712,45 @@ def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
                 else:
                     dim0 = args[1]
                     dim1 = args[2]
-                return g.call_function(ttnn.transpose, args=(args[0], dim0, dim1))
+
+                dim0 = (dim0 + rank) % rank
+                dim1 = (dim1 + rank) % rank
+
+                # directly transpose only support rank <= 4
+                if rank <= 4:
+                    return g.call_function(ttnn.transpose, args=(args[0], dim0, dim1))
+
+                # Try to reshape,
+                #   eg. the original shape [A, B, C, D, E] shape[dim0] = A, shape[dim1] = D
+                #   reshape to [A, (B*C), D, E]
+                originalShape = list(args[0].meta["val"].size())
+                newShape = []
+                newDim0, newDim1 = 0, 1
+                if len(originalShape[:dim0]) > 0:
+                    newShape.append(np.prod(originalShape[:dim0]))
+                    newDim0 += 1
+                    newDim1 += 1
+
+                newShape.append(originalShape[dim0])
+
+                if len(originalShape[dim0 + 1 : dim1]) > 0:
+                    newShape.append(np.prod(originalShape[dim0 + 1 : dim1]))
+                    newDim1 += 1
+
+                newShape.append(originalShape[dim1])
```

Comment:
IIUC here we assume `dim0 < dim1`, maybe need to check and swap if needed

Reply:
Oh yes, It has to check in advance. fixed in 7f572c6eed894451812105f55b72524e501d026e.